diff --git a/hw/xfree86/common/xf86pciBus.c b/hw/xfree86/common/xf86pciBus.c
index 74016af..e40219c 100644
--- a/hw/xfree86/common/xf86pciBus.c
+++ b/hw/xfree86/common/xf86pciBus.c
@@ -1110,8 +1110,10 @@ videoPtrToDriverList(struct pci_device *dev,
 	case 0x8086:
 	    if ((dev->device_id == 0x00d1) || (dev->device_id == 0x7800)) {
 		driverList[0] = "i740";
-            } else if (dev->device_id == 0x8108) {
-                break; /* "hooray" for poulsbo */
+            } else if ((dev->device_id == 0x8108) || (dev->device_id == 0x8109) || (dev->device_id == 0x4102) ||
+		    	(dev->device_id == 0x0130)) {
+		driverList[0] = "pvr";
+		driverList[1] = "psb";
 	    } else {
 		driverList[0] = "intel";
 	    }
